### PR TITLE
ARROW-70 Add Release Notes

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/RELEASE.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+#. Download the release assets from the "Python Wheels" Github Workflow, e.g.
+https://github.com/mongodb-labs/mongo-arrow/actions/runs/2060477840.
+
+#. Upload all the release packages to PyPI with twine::
+
+     $ python3 -m twine upload dist/*
+
+#. Make sure the new version appears on https://mongo-arrow.readthedocs.io/en/latest/. If the
+    new version does not show up automatically, trigger a rebuild of "latest":
```
Comment (ShaneHarvey) [inline bindings/python/RELEASE.rst:52]: The formatting in most of these lines looks off. For example this line needs to be indented only 3 spaces.

File: bindings/python/RELEASE.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#. Ensure that the latest commit is passing CI on GitHub Actions as expected.
+
+#. Check JIRA to ensure all the tickets in this version have been completed.
+
+#. Add release notes to `doc/source/changelog.rst`. Generally just summarize/clarify
+   the git log, but you might add some more long form notes for big changes.
+
+#. Search and replace the `devN` version number w/ the new version number (see
+   note above in `Versioning`_). Make sure version number is updated in
+   `setup.py` and `pymongoarrow/version.py`. Commit the change and tag the release.
```
Comment (ShaneHarvey) [inline bindings/python/RELEASE.rst:33]: pymongoarrow single sources the version in pymongoarrow/version.py so we only need to update in one place, not in setup.py.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/RELEASE.rst (hunk 1 above)
Comment (blink1073) [reply bindings/python/RELEASE.rst]: Fixed

File: bindings/python/RELEASE.rst (hunk 2 above)
Comment (blink1073) [reply bindings/python/RELEASE.rst]: Updated

File: bindings/python/RELEASE.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+   new version does not show up automatically, trigger a rebuild of "latest":
+   https://readthedocs.org/projects/mongo-arrow/builds/
+
+#. Bump the version number to <next version>.dev0 in version.py,
+   commit, push.
+
+#. Publish the release version in Jira.
+
+#. Announce the release on:
+   https://developer.mongodb.com/community/forums/c/community/release-notes/
```
Comment (ShaneHarvey) [inline bindings/python/RELEASE.rst:61]: This link doesn't work for me, let's use: https://www.mongodb.com/community/forums/c/announcements/driver-releases

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/RELEASE.rst (hunk 3 above)
Comment (blink1073) [reply bindings/python/RELEASE.rst]: Fixed

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**